csharp/src/Ice/ObjectAdapter.cs

```diff
@@ -933,13 +933,12 @@ internal bool isLocal(Reference r)
         }
         else
         {
-            EndpointI[] endpoints = r.getEndpoints();
-
+            // Proxies which have at least one endpoint in common with the published endpoints are considered local.
             lock (_mutex)
             {
                 checkForDeactivation();
 
-                // Proxies which have at least one endpoint in common with the published endpoints are considered local.
+                EndpointI[] endpoints = r.getEndpoints();
```

Comment:
Moved here for consistency with Java and other languages. Reference.getEndpoints() doesn't lock any mutex so it can be in this mutex lock.